src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -21,6 +40,13 @@ public NativeBufferPool(int maxElementsInBuffer)
             _buckets = new NativeBufferBucket<T>[maxBuckets + 1];
         }
 
+        public void Dispose()
```

Comment:
Now that the object supports disposal, what happens if one of the member methods gets called?

I do not see any disposed state checking occurring there? We're likely to NRE in the library class. Better to detect the disposal and throw an `ObjectDisposedException` so the violator understands their error.